File: dataset/output/MISTRAL_codestral-latest/task194/Task194_MISTRAL_codestral_latest.java
// Java code
public class Task194_MISTRAL_codestral_latest {
    public void findSecretWord(String[] words, Master master) {
        int matches(String a, String b) {
            int count = 0;
            for (int i = 0; i < a.length(); i++) {
                if (a.charAt(i) == b.charAt(i)) {
                    count++;
                }
            }
            return count;
        }

        void findSecretWord(List<String> possibleWords) {
            if (possibleWords.isEmpty()) {
                return;
            }
            if (possibleWords.size() == 1) {
                master.guess(possibleWords.get(0));
                return;
            }

            // Choose a word with the least number of matches with other words
            int[] wordCount = new int[possibleWords.size()];
            for (int i = 0; i < possibleWords.size(); i++) {
                for (int j = i + 1; j < possibleWords.size(); j++) {
                    int matches = matches(possibleWords.get(i), possibleWords.get(j));
                    wordCount[i] += matches;
                    wordCount[j] += matches;
                }
            }
            int minCount = Integer.MAX_VALUE;
            int minIndex = 0;
            for (int i = 0; i < wordCount.length; i++) {
                if (wordCount[i] < minCount) {
                    minCount = wordCount[i];
                    minIndex = i;
                }
            }
            String guess = possibleWords.get(minIndex);

            int matches = master.guess(guess);
            if (matches == 6) {
                return;
            }

            // Filter out words that don't match the number of matches with the guess
            List<String> newPossibleWords = new ArrayList<>();
            for (String word : possibleWords) {
                if (matches(word, guess) == matches) {
                    newPossibleWords.add(word);
                }
            }
            findSecretWord(newPossibleWords);
        }

        findSecretWord(Arrays.asList(words));
    }
}

File: dataset/output/MISTRAL_codestral-latest/Task194_MISTRAL_codestral-latest.cpp
// C++ code
class Solution {
public:
    void findSecretWord(vector<string>& words, Master& master) {
        int matches(string a, string b) {
            int count = 0;
            for (int i = 0; i < a.length(); i++) {
                if (a[i] == b[i]) {
                    count++;
                }
            }
            return count;
        }

        void findSecretWord(vector<string>& possibleWords) {
            if (possibleWords.empty()) {
                return;
            }
            if (possibleWords.size() == 1) {
                master.guess(possibleWords[0]);
                return;
            }

            // Choose a word with the least number of matches with other words
            vector<int> wordCount(possibleWords.size(), 0);
            for (int i = 0; i < possibleWords.size(); i++) {
                for (int j = i + 1; j < possibleWords.size(); j++) {
                    int matchesCount = matches(possibleWords[i], possibleWords[j]);
                    wordCount[i] += matchesCount;
                    wordCount[j] += matchesCount;
                }
            }
            int minCount = *min_element(wordCount.begin(), wordCount.end());
            string guess = possibleWords[distance(wordCount.begin(), find(wordCount.begin(), wordCount.end(), minCount))];

            int matchesCount = master.guess(guess);
            if (matchesCount == 6) {
                return;
            }

            // Filter out words that don't match the number of matches with the guess
            vector<string> newPossibleWords;
            for (string word : possibleWords) {
                if (matches(word, guess) == matchesCount) {
                    newPossibleWords.push_back(word);
                }
            }
            findSecretWord(newPossibleWords);
        }

        findSecretWord(words);
    }
};